lib/cli/deploy.rs

```diff
@@ -123,3 +98,38 @@ pub fn new_transfer(
         .map_err(crate::Error::from)?;
     Ok(deploy)
 }
+
+fn get_maybe_secret_key(
```

Comment:
Well mostly this code was repeated two times in the file and we already discussed this condition 

```
 let maybe_secret_key = if allow_unsigned_deploy && deploy_params.secret_key.is_empty() {
        None
    } else if deploy_params.secret_key.is_empty() && !allow_unsigned_deploy {
   
 ````
 
 is imo more idiomatic in if A / else if B / else. It's quite complicate to have cargo check and cargo clippy accepting overloads, I tried to have 1/2 methods with both config accordingly and that never passed cargo check/clippy. Most of times the only accepted way is to simply have #[cfg] working like if statements. Otherwise most of the times a function will be missing on compilation if you have like --features-all

Reply:
The structure of the if else is fine, It's functionally equivalent and should work fine.
I do think that if you like the style suggestion I put on the sister PR against the node I would love to see the same style applied to the feature flag here. If you don't want to go with that style, consider the PR approved.

Reply:
Casper client do not seem to have cargo clippy so I can duplicate this method in the client without side effects apparently.

Please check that commit https://github.com/casper-ecosystem/casper-client-rs/commit/4855e7ba3329eb38a960baaf2c62d5fa1121e387 thanks